### ISSUE TITLE
Possible regression error with IAM Role Policy Document

### ISSUE THREAD

**Issue by rpattcorner** [localstack/localstack#2647]
# Type of request: This is a ...

[ X] bug report
[ ] feature request

# Detailed description
Our project's automated tests which ran correctly as recently as 6/22/2020 are consistently failing at the point that we retrieve an IAM role and look into the AssumeRolePolicyDocument.  Rerunning a test marked as successful in GitLab now consistently fails, with no code change in the project.

Investigating, I see that after retrieving a role via boto3:
`role_list = self.account_client.list_roles(PathPrefix='/' + asset_path + '/')["Roles"]`

the AssumeRolePolicy document is no longer a proper string, but is instead a coded representation of the string:
![image](https://user-images.githubusercontent.com/1488518/86050389-c9c46580-ba21-11ea-8f8b-46c96921505a.png)

and failure proceeds from there.  Here's the actual string that's returned instead of the expected structure:
`%22%2522%252522%25257B%25255C%252522Version%25255C%252522%25253A%252520%25255C%2525222012-10-17%25255C%252522%25252C%252520%25255C%252522Statement%25255C%252522%25253A%252520%25255B%25257B%25255C%252522Effect%25255C%252522%25253A%252520%25255C%252522Allow%25255C%252522%25252C%252520%25255C%252522Principal%25255C%252522%25253A%252520%25257B%25255C%252522AWS%25255C%252522%25253A%252520%25255B%25255D%25257D%25252C%252520%25255C%252522Action%25255C%252522%25253A%252520%25255C%252522sts%25253AAssumeRole%25255C%252522%25257D%25255D%25257D%252522%2522%22`

It looks like there's been a [commit recently](https://github.com/localstack/localstack/pull/2610) that affects the AssumeRolePolicy, and wonder (without evidence other than timing and area of code) if there might have been an undesirable side effect.

Another possibility ... because we run localstack in pytest using the [pytest-localstack](https://github.com/mintel/pytest-localstack) framework there's the chance things are now out of sync with the change.  As pytest-localstack isn't updated as frequently as localstack, if this is the likely problem it would be helpful if there are pointers on other ways to run localstack in a pytest framework

## Expected behavior
Expect role retrieval to produce a proper string in AssumeRolePolicy along the following lines (this is a run against actual AWS using same code):
![image](https://user-images.githubusercontent.com/1488518/86051679-cc27bf00-ba23-11ea-9410-cde4a34a8d4d.png)

As you see, the AssumeRolePolicy is represented as an ordinary python structure instead of the coded string

## Actual behavior
Role retrieval yields a structure in expected behaviour

# Steps to reproduce
1. Acquire an IAM boto3 client in mock
2. Get some roles and look at them
3. Either observe the AssumeRolePolicyDocument in a debugger or print, or attempt to access it and fail because the structure returned is corrupt

## Command used to start LocalStack
Started via pytest-localstack
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```
        role_list = .account_client.list_roles(PathPrefix='/' + asset_path + '/')["Roles"]
        role_dict = {}
        for role in role_list:  # Format assume_principals as a list to enable comparisons and add role to dict
            assume_principals = role["AssumeRolePolicyDocument"]["Statement"][0]["Principal"]["AWS"]
```

**Comment by rpattcorner** [comment]
I've got some more insight into the situation.  I can reproduce the issue outside of any reference to pytest-localstack which should narrow our scope, and can describe the issue better and illustrate it simply.

Per the initial report, this is an entirely new behaviour affecting unchanged production code and I'm puzzled where it could be coming from.  Feel free to reassign if need be, or point me at where I've gone wrong.

The issue seems to be that after an initial successful role creation, a subsequent role creation corrupts the AssumeRolePolicyDocument of an existing role.

Reference this test case, which successfully creates a role, illustrates that its ARP is well-formed, then creates a second role and illustrates corruption in the first role:

```
import localstack_client.session

session = localstack_client.session.Session()
iam = session.client('iam')
assume_role_policy_doc = '{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}'

# Clean house
role_list = iam.list_roles()["Roles"]
for role in role_list:
    response = iam.delete_role(RoleName=role['RoleName'])

# Create an initial role
response = iam.create_role(
    Path='path',
    RoleName='fooRole',
    AssumeRolePolicyDocument='{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}',
    Description='string'
)

# Fetch it back and print arp
role_list = iam.list_roles()["Roles"]

print("After creating role 1 as initial sole role, valid ARP")
for role in role_list:
    print ("Role {} has an assumeRolePolicy of {}".format(role['RoleName'],role["AssumeRolePolicyDocument"]))

# Now make another
response = iam.create_role(
    Path='path',
    RoleName='feeRole',
    AssumeRolePolicyDocument='{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}',
    Description='string'
)

role_list = iam.list_roles()["Roles"]

print ("Observe how creating a second role has corrupted the first")
for role in role_list:
    print ("Role {} has an assumeRolePolicy of {}".format(role['RoleName'],role["AssumeRolePolicyDocument"]))


```

With output that looks like this:
```
After creating role 1 as initial sole role, valid ARP
Role fooRole has an assumeRolePolicy of {"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}
Observe how creating a second role has corrupted the first
Role fooRole has an assumeRolePolicy of **%22%7B%5C%22Version%5C%22%3A%20%5C%222012-10-17%5C%22%2C%20%5C%22Statement%5C%22%3A%20%5B%7B%5C%22Effect%5C%22%3A%20%5C%22Allow%5C%22%2C%20%5C%22Principal%5C%22%3A%20%7B%5C%22AWS%5C%22%3A%20%5B%5C%22arn%3Aaws%3Aiam%3A%3A123412341234%3Aroot%5C%22%5D%7D%2C%20%5C%22Action%5C%22%3A%20%5C%22sts%3AAssumeRole%5C%22%7D%5D%7D%22**
Role feeRole has an assumeRolePolicy of {"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}

Process finished with exit code 0
```

**Comment by rpattcorner** [comment]
@duongpv7 Thank you so much for getting to this so quickly.  I may need to wait for the merge to rettest, as I've no idea how to build a localstack from the branch, but will do so as soon as possible.

**Comment by whummer** [comment]
Thanks for the detailed report @rpattcorner . This should be fixed in #2657 - can you do a `docker pull localstack/localstack` and give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks

**Comment by rpattcorner** [comment]
@whummer Works in a local workstation docker after the update, so encouraging, thanks!  Still failing in GitLab CICD because I have to figure out how to get a more recent version working in their runner, not your problem.  But it will probably resolve itself when you push an official tagged version, so it would be great to know how often that happens
